sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -489,7 +515,7 @@ extension StorageBlobClient: TransferDelegate {
     public func transfer(
         _ transfer: Transfer,
         didUpdateWithState state: TransferState,
-        andProgress progress: TransferProgress?
+        andProgress progress: Float?
```

Comment:
`TransferProgress` still exists for raw operations. The TransferManager ones use plain ol' `Float`.

Reply:
I was trying to think a little more about this. Is there a reason we don't want the customer to get a TransferProgress passed to their delegate methods? I could see places where they want to report on the bytes transferred / total bytes. Sure they can cast to BlobTransfer and read from there, but what if it's a MultiBlobTransfer?

Also I was concerned about comparison of Floats. I don't know about Swift, but in some languages comparing floats for equality can lead to bugs because of precision issues, and determining if they differ by some epsilon value is preferred in these cases:
```
if (a == b) // BAD
if (b - a <= Float.EPSILON) // GOOD
```

Reply:
I think it's a valid point. At the time, I got rid of it because we were trying to get rid of relying on the uploader/downloader classes, and creating a `TransferProgress` object had such a dependency. Also, I found I was creating the progress objects for the TransferDelegate calls by passing in completed blocks and total blocks for bytes and total bytes as an imprecise proxy anyways. Since we now recreate the uploader and downloader, there's no reason we couldn't update this to use TransferProgress again. I would want to do it as a separate PR though.

To your point about Float comparisons, the `TransferProgress` struct has the same issues. You can return the progress `asFloat` or `asPercent` (which is just `asFloat * 100`). Float was also chosen because that is the native type for the UIProgressView component.

I would not be surprised if Swift is subject to the same floating point precision errors as other languages, but in my mind this falls squarely on the developer's shoulder, who should know that performing equality comparisons on floating point values is generally unsafe. Most typically, they would be taking action base on inequality operators (i.e. do X when progress > 0.5). What alternative would we have? Even with access to the cumulative and total bytes, they'd inevitably be doing that calculation to a float point value themselves.

Reply:
Right, I'm not sure how important it is, but if for some reason a developer wrote `if progress == 1.0` they might not realize that the comparison may not work like they intend. I do agree it's ultimately the developer's responsibility to know about this, but with a TransferProgress they could check if `bytes == totalBytes`.

The use case for this is probably small though, because really they should be implementing whatever logic in `transterDidComplete` instead.

Reply:
I'm okay with considering reverting this back to `TransferProgress` in a future PR.